flaky: WorkspacePage exceeded timeout of 5000 ms for a test

Spotted in: https://github.com/coder/coder/actions/runs/3629622036/jobs/6122043573

```
Summary of all failing tests
FAIL src/pages/WorkspacePage/WorkspacePage.test.tsx (33.78 s)
  ● WorkspacePage › requests a delete job when the user presses Delete and confirms

    thrown: "Exceeded timeout of 5000 ms for a test.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      87 |
      88 | describe("WorkspacePage", () => {
    > 89 |   it("requests a delete job when the user presses Delete and confirms", async () => {
         |   ^
      90 |     const user = userEvent.setup()
      91 |     const deleteWorkspaceMock = jest
      92 |       .spyOn(api, "deleteWorkspace")

      at src/pages/WorkspacePage/WorkspacePage.test.tsx:89:3
      at Object.<anonymous> (src/pages/WorkspacePage/WorkspacePage.test.tsx:88:1)


Test Suites: 1 failed, 66 passed, 67 total
Tests:       1 failed, 930 passed, 931 total
Snapshots:   0 total
Time:        100.145 s
Ran all test suites.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
Error: Process completed with exit code 1.
```

--- a/site/src/pages/WorkspacePage/WorkspacePage.test.tsx
+++ b/site/src/pages/WorkspacePage/WorkspacePage.test.tsx
@@ -163,9 +163,7 @@ describe("WorkspacePage", () => {
       name: "cancel action",
     })
 
-    await userEvent
-      .setup()
-      .click(cancelButton)
+    await userEvent.setup().click(cancelButton)
 
     expect(cancelWorkspaceMock).toBeCalled()
   })
@@ -185,9 +183,7 @@ describe("WorkspacePage", () => {
     await renderWorkspacePage()
     const buttonText = t("actionButton.update", { ns: "workspacePage" })
     const button = await screen.findByText(buttonText, { exact: true })
-    await userEvent
-      .setup()
-      .click(button)
+    await userEvent.setup().click(button)
 
     // getTemplate is called twice: once when the machine starts, and once after the user requests to update
     expect(getTemplateMock).toBeCalledTimes(2)
@@ -209,9 +205,7 @@ describe("WorkspacePage", () => {
     await renderWorkspacePage()
     const buttonText = t("actionButton.update", { ns: "workspacePage" })
     const button = await screen.findByText(buttonText, { exact: true })
-    await userEvent
-      .setup()
-      .click(button)
+    await userEvent.setup().click(button)
 
     await waitFor(() =>
       expect(api.startWorkspace).toBeCalledWith(

--- a/site/src/pages/WorkspacePage/WorkspacePage.test.tsx
+++ b/site/src/pages/WorkspacePage/WorkspacePage.test.tsx
@@ -1,4 +1,4 @@
-import { fireEvent, screen, waitFor, within } from "@testing-library/react"
+import { screen, waitFor, within } from "@testing-library/react"
 import userEvent from "@testing-library/user-event"
 import EventSourceMock from "eventsourcemock"
 import i18next from "i18next"
@@ -48,9 +48,11 @@ const renderWorkspacePage = async () => {
  * workspaceStatus was calculated correctly.
  */
 const testButton = async (label: string, actionMock: jest.SpyInstance) => {
+  const user = userEvent.setup()
+
   await renderWorkspacePage()
   const button = await screen.findByRole("button", { name: label })
-  fireEvent.click(button)
+  await user.click(button)
   expect(actionMock).toBeCalled()
 }
 
@@ -65,7 +67,7 @@ const testStatus = async (ws: Workspace, label: string) => {
   )
   await renderWorkspacePage()
   const header = screen.getByTestId("header")
-  const status = await within(header).findByRole("status")
+  const status = within(header).getByRole("status")
   expect(status).toHaveTextContent(label)
 }
 
@@ -88,6 +90,7 @@ afterAll(() => {
 describe("WorkspacePage", () => {
   it("requests a delete job when the user presses Delete and confirms", async () => {
     const user = userEvent.setup()
+
     const deleteWorkspaceMock = jest
       .spyOn(api, "deleteWorkspace")
       .mockResolvedValueOnce(MockWorkspaceBuild)
@@ -160,7 +163,9 @@ describe("WorkspacePage", () => {
       name: "cancel action",
     })
 
-    fireEvent.click(cancelButton)
+    await userEvent
+      .setup()
+      .click(cancelButton)
 
     expect(cancelWorkspaceMock).toBeCalled()
   })
@@ -180,7 +185,9 @@ describe("WorkspacePage", () => {
     await renderWorkspacePage()
     const buttonText = t("actionButton.update", { ns: "workspacePage" })
     const button = await screen.findByText(buttonText, { exact: true })
-    fireEvent.click(button)
+    await userEvent
+      .setup()
+      .click(button)
 
     // getTemplate is called twice: once when the machine starts, and once after the user requests to update
     expect(getTemplateMock).toBeCalledTimes(2)
@@ -202,7 +209,9 @@ describe("WorkspacePage", () => {
     await renderWorkspacePage()
     const buttonText = t("actionButton.update", { ns: "workspacePage" })
     const button = await screen.findByText(buttonText, { exact: true })
-    fireEvent.click(button)
+    await userEvent
+      .setup()
+      .click(button)
 
     await waitFor(() =>
       expect(api.startWorkspace).toBeCalledWith(
@@ -211,6 +220,7 @@ describe("WorkspacePage", () => {
       ),
     )
   })
+
   it("shows the Stopping status when the workspace is stopping", async () => {
     await testStatus(
       MockStoppingWorkspace,

